pkg/inject/config_override.go

```diff
@@ -0,0 +1,345 @@
+package inject
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	l5dcharts "github.com/linkerd/linkerd2/pkg/charts/linkerd2"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/linkerd/linkerd2/pkg/util"
+	log "github.com/sirupsen/logrus"
+	k8sResource "k8s.io/apimachinery/pkg/api/resource"
+)
+
+type ConfigLookup interface {
+	GetConfigOverride(string) (string, bool)
```

Comment:
As far as I can tell, this just depends on 3  annotation maps: workload annotations, owner annotations, and namespace annotations.  Therefore this could be a freestanding function that takes 3 annotation maps and a key as inputs.

Reply:
that was my first instinct -- this applies to most of your comments actually -- but I found the API here to be a bit cumbersome. It's just a bit of a shame that we have to bog down the exported functions with concrete bits of information that are coupled to the way `resourceConfig` does things.

For example, in this case we'd think we pass in owner annotations, but we actually pass in a set of annotations that _might_ belong to an object other than a pod, which can really only be a service, since it's the only other object allowed by an injector. We'd actually never want owner annotations to be applied since we don't normally do that for a pod in the cluster. Just feels like a bit of a leaky abstraction to me, but I think that's true regardless of what we end up doing, so we might as well simplify and get rid of the indirection.